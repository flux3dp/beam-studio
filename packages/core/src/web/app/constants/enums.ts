export enum OptionValues {
  FALSE = 'FALSE',
  TRUE = 'TRUE',
}

export default {
  OptionValues,
};
