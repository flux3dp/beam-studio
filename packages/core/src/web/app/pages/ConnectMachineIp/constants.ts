// ConnectMachineIp/constants.ts
export const MACHINE_CONNECTION_TIMEOUT = 30;
