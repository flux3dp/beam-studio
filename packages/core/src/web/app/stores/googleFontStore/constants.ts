export const RETRY_JITTER_MAX = 1000;
export const MAX_CONCURRENT_LOADS = 5;
export const NETWORK_STATE_CHECK_INTERVAL = 30000;
export const CSS_CLEANUP_INTERVAL = 300000;

export const DEFAULT_FONT_WEIGHT = 400;
export const FONT_HISTORY_MAX_SIZE = 5;
export const INITIAL_USAGE_COUNT = 1;
export const QUEUE_PROCESS_DELAY = 100;

export const WEIGHT_FALLBACK_ORDER = [400, 500, 300, 600, 200, 700, 100, 800, 900] as const;

export const PRIORITY_ORDER = { critical: 0, high: 1, low: 3, normal: 2 } as const;

export const ICON_FONT_KEYWORDS = ['icons'] as const;

export const DEFAULT_FALLBACK_FONT = 'Arial, sans-serif';

export const WEB_SAFE_FONTS = [
  'Arial',
  'Arial Black',
  'Comic Sans MS',
  'Courier New',
  'Georgia',
  'Helvetica',
  'Impact',
  'Lucida Console',
  'Lucida Sans Unicode',
  'Palatino Linotype',
  'Tahoma',
  'Times New Roman',
  'Trebuchet MS',
  'Verdana',
] as const;

export const FALLBACK_POSTSCRIPT_NAMES: Record<string, string> = {
  Arial: 'ArialMT',
  'Arial Black': 'Arial-Black',
  'Comic Sans MS': 'ComicSansMS',
  'Courier New': 'CourierNewPSMT',
  Georgia: 'Georgia',
  Helvetica: 'Helvetica',
  Impact: 'Impact',
  'Lucida Console': 'LucidaConsole',
  'Lucida Sans Unicode': 'LucidaSansUnicode',
  'Palatino Linotype': 'PalatinoLinotype',
  Tahoma: 'Tahoma',
  'Times New Roman': 'TimesNewRomanPSMT',
  'Trebuchet MS': 'TrebuchetMS',
  Verdana: 'Verdana',
} as const;

export const WEIGHT_STYLES: Record<number, string> = {
  100: 'Thin',
  200: 'ExtraLight',
  300: 'Light',
  400: 'Regular',
  500: 'Medium',
  600: 'SemiBold',
  700: 'Bold',
  800: 'ExtraBold',
  900: 'Black',
} as const;
