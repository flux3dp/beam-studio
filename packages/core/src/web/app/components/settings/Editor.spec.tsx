import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import { OptionValues } from 'app/constants/enums';

import Editor from './Editor';

jest.mock('helpers/is-dev', () => () => true);

const getFontOfPostscriptName = jest.fn();
jest.mock('app/actions/beambox/font-funcs', () => ({
  requestAvailableFontFamilies: () => ['Arial', 'Courier', 'Apple LiSung'],
  fontNameMap: new Map(
    Object.entries({
      Arial: 'Arial',
      Courier: 'Courier',
      'Apple LiSung': '蘋果儷細宋',
    })
  ),
  requestFontsOfTheFontFamily: (family) => {
    const fonts = {
      Arial: [
        {
          family: 'Arial',
          style: 'Regular',
          postscriptName: 'ArialMT',
        },
        {
          family: 'Arial',
          style: 'Bold',
          postscriptName: 'Arial-BoldMT',
        },
        {
          family: 'Arial',
          style: 'Bold Italic',
          postscriptName: 'Arial-BoldItalicMT',
        },
        {
          family: 'Arial',
          style: 'Italic',
          postscriptName: 'Arial-ItalicMT',
        },
      ],
      Courier: [
        {
          family: 'Courier',
          style: 'Regular',
          postscriptName: 'Regular',
        },
        {
          family: 'Courier',
          style: 'Bold',
          postscriptName: 'Courier-Bold',
        },
        {
          family: 'Courier',
          style: 'Bold Oblique',
          postscriptName: 'Courier-BoldOblique',
        },
        {
          family: 'Courier',
          style: 'Oblique',
          postscriptName: 'Courier-Oblique',
        },
      ],
      'Apple LiSung': [
        {
          family: 'Apple LiSung',
          style: 'Light',
          postscriptName: 'LiSungLight',
        },
      ],
    };
    return fonts[family];
  },
  getFontOfPostscriptName: (...args) => getFontOfPostscriptName(...args),
}));

const map = new Map();
map.set('default-font', {
  family: 'Arial',
  style: 'Regular',
});
jest.mock('implementations/storage', () => ({
  get: (key) => map.get(key),
  set: (key, value) => map.set(key, value),
}));

jest.mock('app/constants/workarea-constants', () => ({
  getWorkarea: () => ({
    label: 'Beambox',
    width: 400,
    pxWidth: 4000,
    height: 375,
    pxHeight: 3750,
    maxSpeed: 300,
    minSpeed: 0.5,
    minPower: 10,
    vectorSpeedLimit: 20,
  }),
}));

jest.mock(
  'app/components/settings/SelectControl',
  () =>
    ({ id, label, url, onChange, options }: any) =>
      (
        <div>
          mock-select-control id:{id}
          label:{label}
          url:{url}
          options:{JSON.stringify(options)}
          <input className="select-control" onChange={onChange} />
        </div>
      )
);

jest.mock(
  'app/widgets/Unit-Input-v2',
  () =>
    ({ id, unit, min, max, defaultValue, getValue, forceUsePropsUnit, className }: any) =>
      (
        <div>
          mock-unit-input id:{id}
          unit:{unit}
          min:{min}
          max:{max}
          defaultValue:{defaultValue}
          forceUsePropsUnit:{forceUsePropsUnit ? 'true' : 'false'}
          className:{JSON.stringify(className)}
          <input className="unit-input" onChange={(e) => getValue(+e.target.value)} />
        </div>
      )
);

jest.mock('helpers/api/swiftray-client', () => ({
  hasSwiftray: true,
}));

jest.mock('helpers/locale-helper', () => ({
  isTwOrHk: true,
}));

describe('settings/Editor', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  test('initially no warning', async () => {
    const updateConfigChange = jest.fn();
    const updateBeamboxPreferenceChange = jest.fn();
    const updateModel = jest.fn();
    const mockGetBeamboxPreferenceEditingValue = jest.fn();
    mockGetBeamboxPreferenceEditingValue.mockImplementation((key) => {
      if (key === 'guide_x0' || key === 'guide_y0') {
        return 0;
      }
      return false;
    });
    const { container } = render(
      <Editor
        defaultUnit="mm"
        selectedModel="fbb1b"
        getBeamboxPreferenceEditingValue={mockGetBeamboxPreferenceEditingValue}
        updateConfigChange={updateConfigChange}
        updateBeamboxPreferenceChange={updateBeamboxPreferenceChange}
        updateModel={updateModel}
      />
    );
    expect(mockGetBeamboxPreferenceEditingValue).toBeCalledTimes(11);
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(1, 'guide_x0');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(2, 'guide_y0');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(3, 'show_guides');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(4, 'image_downsampling');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(5, 'anti-aliasing');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(6, 'continuous_drawing');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(
      7,
      'simplify_clipper_path'
    );
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(8, 'enable-low-speed');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(9, 'auto-switch-tab');
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(
      10,
      'enable-custom-backlash'
    );
    expect(mockGetBeamboxPreferenceEditingValue).toHaveBeenNthCalledWith(11, 'path-engine');
    expect(container).toMatchSnapshot();

    const SelectControls = container.querySelectorAll('.select-control');
    const UnitInputs = container.querySelectorAll('.unit-input');

    fireEvent.change(SelectControls[0], { target: { value: 'inches' } });
    expect(updateConfigChange).toHaveBeenCalledTimes(1);
    expect(updateConfigChange).toHaveBeenNthCalledWith(1, 'default-units', 'inches');

    fireEvent.change(SelectControls[1], { target: { value: 'Apple LiSung' } });
    expect(container).toMatchSnapshot();

    fireEvent.change(SelectControls[1], { target: { value: 'Courier' } });
    expect(container).toMatchSnapshot();

    getFontOfPostscriptName.mockReturnValue({
      family: 'Courier',
      style: 'Bold',
      postscriptName: 'Courier-Bold',
    });
    fireEvent.change(SelectControls[2], { target: { value: 'Courier-Bold' } });
    expect(container).toMatchSnapshot();

    fireEvent.change(SelectControls[3], { target: { value: 'fbm1' } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(1);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(1, 'model', 'fbm1');
    expect(updateModel).toHaveBeenCalledTimes(1);
    expect(updateModel).toHaveBeenNthCalledWith(1, 'fbm1');

    fireEvent.change(SelectControls[4], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(2);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(2, 'show_guides', 'TRUE');

    fireEvent.change(SelectControls[5], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(3);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(3, 'image_downsampling', 'TRUE');

    fireEvent.change(SelectControls[6], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(4);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(4, 'anti-aliasing', 'TRUE');

    fireEvent.change(SelectControls[7], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(5);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(5, 'continuous_drawing', 'TRUE');

    fireEvent.change(SelectControls[8], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(6);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(
      6,
      'simplify_clipper_path',
      'TRUE'
    );

    fireEvent.change(SelectControls[9], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(7);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(7, 'enable-low-speed', 'TRUE');

    fireEvent.change(SelectControls[10], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(8);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(8, 'auto-switch-tab', 'TRUE');

    fireEvent.change(UnitInputs[0], { target: { value: 1 } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(9);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(9, 'guide_x0', 1);

    fireEvent.change(UnitInputs[1], { target: { value: 2 } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(10);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(10, 'guide_y0', 2);

    fireEvent.change(SelectControls[11], { target: { value: 'swiftray' } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(11);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(11, 'path-engine', 'swiftray');

    fireEvent.change(SelectControls[12], { target: { value: OptionValues.TRUE } });
    expect(updateBeamboxPreferenceChange).toHaveBeenCalledTimes(12);
    expect(updateBeamboxPreferenceChange).toHaveBeenNthCalledWith(
      12,
      'enable-custom-backlash',
      'TRUE'
    );
  });
});
