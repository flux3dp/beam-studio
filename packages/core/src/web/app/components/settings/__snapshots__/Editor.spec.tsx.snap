// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings/Editor initially no warning 1`] = `
<div>
  <div
    class="subtitle"
  >
    Editor
  </div>
  <div>
    mock-select-control id:
    set-default-units
    label:
    Default Units
    url:
    options:
    [{"label":"mm","value":"mm"},{"label":"Inches","value":"inches"}]
    <input
      class="select-control"
      id="set-default-units"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-family
    label:
    Default Font
    url:
    options:
    [{"label":"Arial","value":"Arial"},{"label":"Courier","value":"Courier"},{"label":"蘋果儷細宋","value":"Apple LiSung"}]
    <input
      class="select-control"
      id="set-default-font-family"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-style
    label:
    Default Font Style
    url:
    options:
    [{"label":"Regular","value":"ArialMT"},{"label":"Bold","value":"Arial-BoldMT"},{"label":"Bold Italic","value":"Arial-BoldItalicMT"},{"label":"Italic","value":"Arial-ItalicMT"}]
    <input
      class="select-control"
      id="set-default-font-style"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-model
    label:
    Default Document Setting
    url:
    options:
    [{"label":"beamo","value":"fbm1"},{"label":"Beambox","value":"fbb1b"},{"label":"Beambox Pro","value":"fbb1p"},{"label":"HEXA","value":"fhexa1"},{"label":"HEXA RF","value":"fhx2rf3"},{"label":"Ador","value":"ado1"},{"label":"Promark","value":"fpm1"},{"label":"Lazervida","value":"flv1"},{"label":"Beambox II","value":"fbb2"}]
    <input
      class="select-control"
      id="set-default-model"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide
    label:
    Guides
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-guide"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide-axis
    label:
    Guides Origin
    url:
    options:
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      X
    </span>
    <div>
      mock-unit-input id:
      guide-x-input
      unit:
      mm
      min:
      0
      max:
      400
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      Y
    </span>
    <div>
      mock-unit-input id:
      guide-y-input
      unit:
      mm
      min:
      0
      max:
      375
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
  <div>
    mock-select-control id:
    set-bitmap-quality
    label:
    Bitmap Preview Quality
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004494995
    options:
    [{"label":"Low","value":true},{"label":"Normal","value":false}]
    <input
      class="select-control"
      id="set-bitmap-quality"
    />
  </div>
  <div>
    mock-select-control id:
    set-anti-aliasing
    label:
    Anti-Aliasing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004408956
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-anti-aliasing"
    />
  </div>
  <div>
    mock-select-control id:
    set-continuous-drawing
    label:
    Continuous Drawing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004406496
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-continuous-drawing"
    />
  </div>
  <div>
    mock-select-control id:
    set-simplify-clipper-path
    label:
    Optimize the Calculated Path
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004407276
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-simplify-clipper-path"
    />
  </div>
  <div>
    mock-select-control id:
    auto-switch-tab
    label:
    Auto Switch Layer and Object Panel
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="auto-switch-tab"
    />
  </div>
  <div>
    mock-select-control id:
    path-engine
    label:
    Path Calculation Acceleration (Beta)
    url:
    https://support.flux3dp.com/hc/en-us/articles/11146997425039
    options:
    [{"label":"On","value":"swiftray"},{"label":"Off","value":"fluxghost"}]
    <input
      class="select-control"
      id="path-engine"
    />
  </div>
  <div>
    mock-select-control id:
    set-enable-custom-backlash
    label:
    Enable Custom Backlash Compensation
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-enable-custom-backlash"
    />
  </div>
</div>
`;

exports[`settings/Editor initially no warning 2`] = `
<div>
  <div
    class="subtitle"
  >
    Editor
  </div>
  <div>
    mock-select-control id:
    set-default-units
    label:
    Default Units
    url:
    options:
    [{"label":"mm","value":"mm"},{"label":"Inches","value":"inches"}]
    <input
      class="select-control"
      id="set-default-units"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-family
    label:
    Default Font
    url:
    options:
    [{"label":"Arial","value":"Arial"},{"label":"Courier","value":"Courier"},{"label":"蘋果儷細宋","value":"Apple LiSung"}]
    <input
      class="select-control"
      id="set-default-font-family"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-style
    label:
    Default Font Style
    url:
    options:
    [{"label":"Light","value":"LiSungLight"}]
    <input
      class="select-control"
      id="set-default-font-style"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-model
    label:
    Default Document Setting
    url:
    options:
    [{"label":"beamo","value":"fbm1"},{"label":"Beambox","value":"fbb1b"},{"label":"Beambox Pro","value":"fbb1p"},{"label":"HEXA","value":"fhexa1"},{"label":"HEXA RF","value":"fhx2rf3"},{"label":"Ador","value":"ado1"},{"label":"Promark","value":"fpm1"},{"label":"Lazervida","value":"flv1"},{"label":"Beambox II","value":"fbb2"}]
    <input
      class="select-control"
      id="set-default-model"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide
    label:
    Guides
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-guide"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide-axis
    label:
    Guides Origin
    url:
    options:
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      X
    </span>
    <div>
      mock-unit-input id:
      guide-x-input
      unit:
      mm
      min:
      0
      max:
      400
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      Y
    </span>
    <div>
      mock-unit-input id:
      guide-y-input
      unit:
      mm
      min:
      0
      max:
      375
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
  <div>
    mock-select-control id:
    set-bitmap-quality
    label:
    Bitmap Preview Quality
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004494995
    options:
    [{"label":"Low","value":true},{"label":"Normal","value":false}]
    <input
      class="select-control"
      id="set-bitmap-quality"
    />
  </div>
  <div>
    mock-select-control id:
    set-anti-aliasing
    label:
    Anti-Aliasing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004408956
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-anti-aliasing"
    />
  </div>
  <div>
    mock-select-control id:
    set-continuous-drawing
    label:
    Continuous Drawing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004406496
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-continuous-drawing"
    />
  </div>
  <div>
    mock-select-control id:
    set-simplify-clipper-path
    label:
    Optimize the Calculated Path
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004407276
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-simplify-clipper-path"
    />
  </div>
  <div>
    mock-select-control id:
    auto-switch-tab
    label:
    Auto Switch Layer and Object Panel
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="auto-switch-tab"
    />
  </div>
  <div>
    mock-select-control id:
    path-engine
    label:
    Path Calculation Acceleration (Beta)
    url:
    https://support.flux3dp.com/hc/en-us/articles/11146997425039
    options:
    [{"label":"On","value":"swiftray"},{"label":"Off","value":"fluxghost"}]
    <input
      class="select-control"
      id="path-engine"
    />
  </div>
  <div>
    mock-select-control id:
    set-enable-custom-backlash
    label:
    Enable Custom Backlash Compensation
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-enable-custom-backlash"
    />
  </div>
</div>
`;

exports[`settings/Editor initially no warning 3`] = `
<div>
  <div
    class="subtitle"
  >
    Editor
  </div>
  <div>
    mock-select-control id:
    set-default-units
    label:
    Default Units
    url:
    options:
    [{"label":"mm","value":"mm"},{"label":"Inches","value":"inches"}]
    <input
      class="select-control"
      id="set-default-units"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-family
    label:
    Default Font
    url:
    options:
    [{"label":"Arial","value":"Arial"},{"label":"Courier","value":"Courier"},{"label":"蘋果儷細宋","value":"Apple LiSung"}]
    <input
      class="select-control"
      id="set-default-font-family"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-style
    label:
    Default Font Style
    url:
    options:
    [{"label":"Regular","value":"Regular"},{"label":"Bold","value":"Courier-Bold"},{"label":"Bold Oblique","value":"Courier-BoldOblique"},{"label":"Oblique","value":"Courier-Oblique"}]
    <input
      class="select-control"
      id="set-default-font-style"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-model
    label:
    Default Document Setting
    url:
    options:
    [{"label":"beamo","value":"fbm1"},{"label":"Beambox","value":"fbb1b"},{"label":"Beambox Pro","value":"fbb1p"},{"label":"HEXA","value":"fhexa1"},{"label":"HEXA RF","value":"fhx2rf3"},{"label":"Ador","value":"ado1"},{"label":"Promark","value":"fpm1"},{"label":"Lazervida","value":"flv1"},{"label":"Beambox II","value":"fbb2"}]
    <input
      class="select-control"
      id="set-default-model"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide
    label:
    Guides
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-guide"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide-axis
    label:
    Guides Origin
    url:
    options:
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      X
    </span>
    <div>
      mock-unit-input id:
      guide-x-input
      unit:
      mm
      min:
      0
      max:
      400
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      Y
    </span>
    <div>
      mock-unit-input id:
      guide-y-input
      unit:
      mm
      min:
      0
      max:
      375
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
  <div>
    mock-select-control id:
    set-bitmap-quality
    label:
    Bitmap Preview Quality
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004494995
    options:
    [{"label":"Low","value":true},{"label":"Normal","value":false}]
    <input
      class="select-control"
      id="set-bitmap-quality"
    />
  </div>
  <div>
    mock-select-control id:
    set-anti-aliasing
    label:
    Anti-Aliasing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004408956
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-anti-aliasing"
    />
  </div>
  <div>
    mock-select-control id:
    set-continuous-drawing
    label:
    Continuous Drawing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004406496
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-continuous-drawing"
    />
  </div>
  <div>
    mock-select-control id:
    set-simplify-clipper-path
    label:
    Optimize the Calculated Path
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004407276
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-simplify-clipper-path"
    />
  </div>
  <div>
    mock-select-control id:
    auto-switch-tab
    label:
    Auto Switch Layer and Object Panel
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="auto-switch-tab"
    />
  </div>
  <div>
    mock-select-control id:
    path-engine
    label:
    Path Calculation Acceleration (Beta)
    url:
    https://support.flux3dp.com/hc/en-us/articles/11146997425039
    options:
    [{"label":"On","value":"swiftray"},{"label":"Off","value":"fluxghost"}]
    <input
      class="select-control"
      id="path-engine"
    />
  </div>
  <div>
    mock-select-control id:
    set-enable-custom-backlash
    label:
    Enable Custom Backlash Compensation
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-enable-custom-backlash"
    />
  </div>
</div>
`;

exports[`settings/Editor initially no warning 4`] = `
<div>
  <div
    class="subtitle"
  >
    Editor
  </div>
  <div>
    mock-select-control id:
    set-default-units
    label:
    Default Units
    url:
    options:
    [{"label":"mm","value":"mm"},{"label":"Inches","value":"inches"}]
    <input
      class="select-control"
      id="set-default-units"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-family
    label:
    Default Font
    url:
    options:
    [{"label":"Arial","value":"Arial"},{"label":"Courier","value":"Courier"},{"label":"蘋果儷細宋","value":"Apple LiSung"}]
    <input
      class="select-control"
      id="set-default-font-family"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-font-style
    label:
    Default Font Style
    url:
    options:
    [{"label":"Regular","value":"Regular"},{"label":"Bold","value":"Courier-Bold"},{"label":"Bold Oblique","value":"Courier-BoldOblique"},{"label":"Oblique","value":"Courier-Oblique"}]
    <input
      class="select-control"
      id="set-default-font-style"
    />
  </div>
  <div>
    mock-select-control id:
    set-default-model
    label:
    Default Document Setting
    url:
    options:
    [{"label":"beamo","value":"fbm1"},{"label":"Beambox","value":"fbb1b"},{"label":"Beambox Pro","value":"fbb1p"},{"label":"HEXA","value":"fhexa1"},{"label":"HEXA RF","value":"fhx2rf3"},{"label":"Ador","value":"ado1"},{"label":"Promark","value":"fpm1"},{"label":"Lazervida","value":"flv1"},{"label":"Beambox II","value":"fbb2"}]
    <input
      class="select-control"
      id="set-default-model"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide
    label:
    Guides
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-guide"
    />
  </div>
  <div>
    mock-select-control id:
    set-guide-axis
    label:
    Guides Origin
    url:
    options:
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      X
    </span>
    <div>
      mock-unit-input id:
      guide-x-input
      unit:
      mm
      min:
      0
      max:
      400
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
    <span
      class="font2"
      style="line-height: 32px; margin-right: 10px;"
    >
      Y
    </span>
    <div>
      mock-unit-input id:
      guide-y-input
      unit:
      mm
      min:
      0
      max:
      375
      defaultValue:
      0
      forceUsePropsUnit:
      true
      className:
      {"half":true}
      <input
        class="unit-input"
      />
    </div>
  </div>
  <div>
    mock-select-control id:
    set-bitmap-quality
    label:
    Bitmap Preview Quality
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004494995
    options:
    [{"label":"Low","value":true},{"label":"Normal","value":false}]
    <input
      class="select-control"
      id="set-bitmap-quality"
    />
  </div>
  <div>
    mock-select-control id:
    set-anti-aliasing
    label:
    Anti-Aliasing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004408956
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-anti-aliasing"
    />
  </div>
  <div>
    mock-select-control id:
    set-continuous-drawing
    label:
    Continuous Drawing
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004406496
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-continuous-drawing"
    />
  </div>
  <div>
    mock-select-control id:
    set-simplify-clipper-path
    label:
    Optimize the Calculated Path
    url:
    https://support.flux3dp.com/hc/en-us/articles/360004407276
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-simplify-clipper-path"
    />
  </div>
  <div>
    mock-select-control id:
    auto-switch-tab
    label:
    Auto Switch Layer and Object Panel
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="auto-switch-tab"
    />
  </div>
  <div>
    mock-select-control id:
    path-engine
    label:
    Path Calculation Acceleration (Beta)
    url:
    https://support.flux3dp.com/hc/en-us/articles/11146997425039
    options:
    [{"label":"On","value":"swiftray"},{"label":"Off","value":"fluxghost"}]
    <input
      class="select-control"
      id="path-engine"
    />
  </div>
  <div>
    mock-select-control id:
    set-enable-custom-backlash
    label:
    Enable Custom Backlash Compensation
    url:
    options:
    [{"label":"On","value":true},{"label":"Off","value":false}]
    <input
      class="select-control"
      id="set-enable-custom-backlash"
    />
  </div>
</div>
`;
